test_suite/gap/test_gap_state.py

```diff
@@ -322,3 +331,95 @@ def test_advertising_after_disconnection(peripheral, peripheral_address, central
     scan_records = peripheral.gap.scanForAddress(central_address['address'], test_params.scanTimeout).result
     assert len(scan_records) > 0
 
+
+@pytest.mark.ble41
+def test_extended_advertising(peripheral, peripheral_address, central, central_address, test_params):
+    """test extended advertising start and stop"""
+
+    central_supported = peripheral.gap.isFeatureSupported("LE_EXTENDED_ADVERTISING").result
+    peripheral_supported = central.gap.isFeatureSupported("LE_EXTENDED_ADVERTISING").result
+    if central_supported and peripheral_supported:
+        handle = peripheral.gap.createAdvertisingSet().result
+        peripheral.advParams.setType("NON_CONNECTABLE_UNDIRECTED")
+        peripheral.gap.setAdvertisingParameters(handle)
+
+        peripheral.gap.startAdvertising(handle, ADV_DURATION_FOREVER, ADV_MAX_EVENTS_UNLIMITED)
+        sleep(1)
```

Comment:
Why do we need to sleep ? It seems that `startAdvertising` blocks until `onAdvertisingStart` is called.
See https://github.com/ARMmbed/mbed-os-bluetooth-integration-testsuite/commit/8694712d23b8443697e8d4a90e50b0390dd760b8 .